cloudinit/sources/DataSourceScaleway.py

```diff
@@ -177,51 +186,152 @@ def __init__(self, sys_cfg, distro, paths):
         self.ds_cfg = util.mergemanydict(
             [
                 util.get_cfg_by_path(sys_cfg, ["datasource", "Scaleway"], {}),
-                BUILTIN_DS_CONFIG,
             ]
         )
 
-        self.metadata_address = self.ds_cfg["metadata_url"]
-        self.userdata_address = self.ds_cfg["userdata_url"]
-        self.vendordata_address = self.ds_cfg["vendordata_url"]
-
         self.retries = int(self.ds_cfg.get("retries", DEF_MD_RETRIES))
         self.timeout = int(self.ds_cfg.get("timeout", DEF_MD_TIMEOUT))
+        self.alive = int(self.ds_cfg.get("alive", DEF_MD_ALIVE))
         self._fallback_interface = None
         self._network_config = sources.UNSET
+        self.metadata_urls = DS_BASE_URLS
+        self.userdata_url = None
+        self.vendordata_url = None
+        self.ephemeral_fixed_address = None
+        self.has_ipv4 = True
+        if "metadata_urls" in self.ds_cfg.keys():
+            # Merge and deduplicate lists
+            self.metadata_urls = list(
+                set(self.metadata_urls) | set(self.ds_cfg["metadata_urls"])
+            )
+
+    def _set_metadata_url(self, urls):
+        """
+        Define metadata_url based upon api-metadata URL availability.
+        """
+
+        for url in urls:
+            resp = None
+            for _ in range(0, self.retries):
+                try:
+                    LOG.debug("Trying to reach %s", url)
+                    resp = requests.get(
+                        url + "/conf?format=json",
+                        timeout=self.alive,
+                    )
+                except ConnectionError:
+                    continue
+                break
+            if not resp:
+                continue
+            if resp.status_code == 200:
+                LOG.debug("%s is reachable", url)
+                self.metadata_url = f"{url}/conf?format=json"
+                self.userdata_url = f"{url}/user_data/cloud-init"
+                self.vendordata_url = f"{url}/vendor_data/cloud-init"
+                return
+        LOG.debug("Unable to reach api-metadata at %s", urls)
+        raise ConnectionError
 
     def _crawl_metadata(self):
         resp = url_helper.readurl(
-            self.metadata_address, timeout=self.timeout, retries=self.retries
+            self.metadata_url, timeout=self.timeout, retries=self.retries
         )
         self.metadata = json.loads(util.decode_binary(resp.contents))
 
         self.userdata_raw = query_data_api(
-            "user-data", self.userdata_address, self.retries, self.timeout
+            "user-data", self.userdata_url, self.retries, self.timeout
         )
         self.vendordata_raw = query_data_api(
-            "vendor-data", self.vendordata_address, self.retries, self.timeout
+            "vendor-data", self.vendordata_url, self.retries, self.timeout
         )
 
+    def _set_urls_on_ip_version(self, proto, urls):
+
+        if proto not in ["ipv4", "ipv6"]:
```

Comment:
Why? Since all of the call sites currently have known values for `proto`, this won't ever get hit (except in the unit test).

Reply:
No longer relevant as this method has been dropped.